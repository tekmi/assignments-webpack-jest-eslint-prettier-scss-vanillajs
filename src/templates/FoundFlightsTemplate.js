const flightTemplate = flight => `
<li class="rw-autosuggest__results-item" data-ft-element="flights-search-results-found">
  <span class="rw-autosuggest-group__flightnumber">${flight.flightNumber}</span>
  <span class="rw-autosuggest-group__airport">${flight.airport}</span>
  <div class="rw-autosuggest-group__time">
    <time>${flight.expectedTime}</time>
  </div>
</li>
`;

export default flights => `
<li class="rw-autosuggest__results-flights">
  <p class="rw-autosuggest__title">Looking for this flight?</p>
  <ul class="rw-autosuggest-group">
    ${flights && flights.map(flightTemplate)}
  </ul>
</li>
`;
