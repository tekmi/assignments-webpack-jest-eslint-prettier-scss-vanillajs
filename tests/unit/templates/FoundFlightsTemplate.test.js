import FoundFlightsTemplate from "../../../src/templates/FoundFlightsTemplate";

describe("FoundFlightsTemplate", () => {
  it("renders empty results group if none flights provided", () => {
    expect(FoundFlightsTemplate([])).toMatchSnapshot();
    expect(FoundFlightsTemplate(null)).toMatchSnapshot();
  });

  it("renders results group with empty flight", () => {
    expect(FoundFlightsTemplate([{}])).toMatchSnapshot();
  });

  it("renders results group with all flight information properly", () => {
    expect(
      FoundFlightsTemplate([
        {
          flightNumber: "UA 989",
          airport: "San Francisco",
          expectedTime: "14:50",
        },
      ]),
    ).toMatchSnapshot();
  });
});
