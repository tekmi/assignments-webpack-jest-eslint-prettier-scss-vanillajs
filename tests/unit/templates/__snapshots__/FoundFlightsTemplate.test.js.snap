// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FoundFlightsTemplate renders empty results group if none flights provided 1`] = `
"
<li class=\\"rw-autosuggest__results-flights\\">
  <p class=\\"rw-autosuggest__title\\">Looking for this flight?</p>
  <ul class=\\"rw-autosuggest-group\\">
    
  </ul>
</li>
"
`;

exports[`FoundFlightsTemplate renders empty results group if none flights provided 2`] = `
"
<li class=\\"rw-autosuggest__results-flights\\">
  <p class=\\"rw-autosuggest__title\\">Looking for this flight?</p>
  <ul class=\\"rw-autosuggest-group\\">
    null
  </ul>
</li>
"
`;

exports[`FoundFlightsTemplate renders results group with all flight information properly 1`] = `
"
<li class=\\"rw-autosuggest__results-flights\\">
  <p class=\\"rw-autosuggest__title\\">Looking for this flight?</p>
  <ul class=\\"rw-autosuggest-group\\">
    
<li class=\\"rw-autosuggest__results-item\\" data-ft-element=\\"flights-search-results-found\\">
  <span class=\\"rw-autosuggest-group__flightnumber\\">UA 989</span>
  <span class=\\"rw-autosuggest-group__airport\\">San Francisco</span>
  <div class=\\"rw-autosuggest-group__time\\">
    <time>14:50</time>
  </div>
</li>

  </ul>
</li>
"
`;

exports[`FoundFlightsTemplate renders results group with empty flight 1`] = `
"
<li class=\\"rw-autosuggest__results-flights\\">
  <p class=\\"rw-autosuggest__title\\">Looking for this flight?</p>
  <ul class=\\"rw-autosuggest-group\\">
    
<li class=\\"rw-autosuggest__results-item\\" data-ft-element=\\"flights-search-results-found\\">
  <span class=\\"rw-autosuggest-group__flightnumber\\">undefined</span>
  <span class=\\"rw-autosuggest-group__airport\\">undefined</span>
  <div class=\\"rw-autosuggest-group__time\\">
    <time>undefined</time>
  </div>
</li>

  </ul>
</li>
"
`;
